python/python/lancedb/namespace.py

```diff
@@ -237,6 +249,19 @@ def table_names(
         *,
         namespace: Optional[List[str]] = None,
     ) -> Iterable[str]:
+        """
+        List table names in the database.
+
+        .. deprecated::
+            Use :meth:`list_tables` instead, which provides proper pagination support.
+        """
+        import warnings
+
+        warnings.warn(
+            "table_names() is deprecated, use list_tables() instead",
+            DeprecationWarning,
+            stacklevel=2,
+        )
```

Comment:
Do we have a migration guide for lancedb?  It would be good to mention this.

Reply:
Created https://github.com/lancedb/lancedb/issues/2840 tracking this